src/extension.ts

```diff
@@ -230,7 +234,13 @@ async function activateServerForFolder(context: ExtensionContext, uri: Uri, fold
   }
   logger.info(cwdMsg);
 
-  const serverEnvironment: IEnvVars = workspace.getConfiguration('haskell', uri).serverEnvironment;
+  let serverEnvironment: IEnvVars = workspace.getConfiguration('haskell', uri).serverEnvironment;
+  if (addInternalServerPath != undefined) {
+      const pathSep = process.platform === 'win32' ? ';' : ':';
+      const PATH = process.env.PATH!.split(pathSep);
+      PATH.unshift(addInternalServerPath);
+      serverEnvironment["PATH"] = PATH.join(pathSep);
+  }
```

Comment:
Somehow this doesn't work... I tried adding prints and somewhere here it seems to silently crash or whatever. I don't have the patience to investigate this js nonsense.

Installation itself seems to work, but now we want HLS to see the rest of the binaries where `haskell-language-wrapper` resides.

Reply:
What other binaries? The extension should only need the path to the HLS-wrapper executable and the path to the HLS executable.

Reply:
```
stack trace: Error: TypeError: Cannot assign to read only property 'PATH' of object
	at Object.set (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:97:44518)
	at /home/hugin/Documents/haskell/vscode-haskell/dist/extension.js:212:39
	at Generator.next (<anonymous>)
	at fulfilled (/home/hugin/Documents/haskell/vscode-haskell/dist/extension.js:11:58)
```
I can help out with the extension a bit

Reply:
```
stack trace: Error: TypeError: Cannot assign to read only property 'PATH' of object
	at Object.set (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:97:44518)
	at /home/hugin/Documents/haskell/vscode-haskell/dist/extension.js:212:39
	at Generator.next (<anonymous>)
	at fulfilled (/home/hugin/Documents/haskell/vscode-haskell/dist/extension.js:11:58)
```
I can help out with the extension a bit

Reply:
```suggestion
  let serverEnvironment: IEnvVars = workspace.getConfiguration('haskell', uri).serverEnvironment;
  if (addInternalServerPath != undefined) {
      const pathSep = process.platform === 'win32' ? ';' : ':';
      const PATH = process.env.PATH!.split(pathSep);
      PATH.unshift(addInternalServerPath);
      serverEnvironment = {
        PATH: PATH.join(pathSep),
        ... serverEnvironment
      };
  }
```

This seems to work

Reply:
BTW, you can see errors of your extension by looking at "Debug Console" of the original VSCode Window (the one that contains the extension's code)

Reply:
> What other binaries? The extension should only need the path to the HLS-wrapper executable and the path to the HLS executable.

Yes, but the path to the hls-wrapper is not in PATH. It didn't work here.

Reply:
> This seems to work

Nice, that works indeed.